lib/ood_core/job/adapters/kubernetes/templates/pod.yml.erb

```diff
@@ -7,6 +7,7 @@ metadata:
     job: <%= id %>
     app.kubernetes.io/name: <%= container.name %>
     app.kubernetes.io/managed-by: open-ondemand
+    account: <%= script.accounting_id %>
```

Comment:
We should probably account for nil here. Otherwise it would be `account: ` which would probably throw a yaml parsing error by kubectl which is particularly unreadable to users.

Reply:
I updated to guard against `nil` `account_id`.

Reply:
Cool. I pulled the other PRs in (Thank you for them!).  I just need to test this to be sure that `account: -` works.

Reply:
Why would there be a hyphen for `account`?  Also for at least nodes empty labels are valid so I would imagine empty labels are fine. In Go an undefined string is an empty string and if their labels are `map[string]string` then no value is same as `key: ""`.

Reply:
I think any valid string for the value will be fine assuming the Go code for labels defines it as `map[string]string`, I think the only way it would not work is if they define labels as `map[string]interface` but I doubt they do that.

Reply:
Am I misreading `ERB.new(File.read(resource_file), nil, '-')`? I read that as replace `nil` values with `-`. 

In any case, the comment was just to update you that I want to put it through an end to end test just to be sure.

Reply:
That `'-'` in `ERB.new` instructs ERB rendering to strip newlines when `-%>` is used and strip leading spaces with `<%-` is used.  See the change to test rendered YAML, all the extra newlines are no longer present. So that something like this:

```
   <%- if foo -%>
bar
   <%- end -%>
```

WIll render just `bar` and not:

```
   
bar

```